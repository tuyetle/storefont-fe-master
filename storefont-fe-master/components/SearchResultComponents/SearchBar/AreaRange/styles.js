import { css } from 'glamor';
import variables from '~/styles/variables';

export default {
  areaRange: css({
    width: '34em',
    maxWidth: '598px',
    padding: '0 0 0.5em',
    top: '0em',
    right: '-11em',
    backgroundColor: 'white',

    '& .input-group': {
      width: '190px',
      height: '30px',
      border: `0.5px solid ${variables.$borderPrice}`,
      fontSize: '0.875em',
      position: 'relative',
      '& .form-group': {
        position: 'relative',
        width: '100%',
      },
      '& .form-control': {
        background: 'none',
        fontSize: '0.875em',
        border: '0',
        fontStyle: 'italic',
        position: 'absolute',
        width: '80%',
        left: '0',
        top: '0',
        lineHeight: '21px',
        '&:focus': {
          outline: 'none',
          boxShadow: 'none',
        },
      },
      '& .input-group-addon': {
        position: 'absolute',
        right: '0',
        background: 'none',
        border: '0',
        fontSize: '1em',
        lineHeight: '18px',
        color: variables.$bgLineMenu,
      },
    },

    '@media (min-width: 768px)': {
      position: 'absolute',
      width: '34em',
      maxWidth: '598px',
      padding: '1.5em 0 0.5em 1.2em',
      top: '0em',
      right: '-10.6em',
      backgroundColor: 'white',
      boxShadow: variables.$boxShadowPrice,
      border: `1px solid ${variables.$borderFilter}`,
    },
  }),
  paddingLeft: css({
    verticalAlign: 'middle',
    paddingLeft: '1em',
    marginLeft: '0',
    marginTop: '1em',
    '@media (min-width: 768px)': {
      marginLeft: '-2em',
      marginTop: '0',
    },
  }),
  priceLabel: css({
    float: 'left',
    marginRight: '0.8em',
    fontSize: '0.875em',
    color: variables.$colorWelcome,
    marginTop: '-2.5px',
    width: '2.5em',
    textAlign: 'left',
    '@media (min-width: 768px)': {
      width: 'auto',
      textAlign: 'auto',
    },
  }),
  error: css({
    textAlign: 'left',
    clear: 'both',
    lineHeight: '1.5em',
    color: `${variables.$errorColor}`,
    maxWidth: '21.563em',
  }),
};
