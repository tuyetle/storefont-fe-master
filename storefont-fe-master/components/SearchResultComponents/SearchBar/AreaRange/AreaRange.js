import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Field } from 'react-form';
import { NumericFormField } from '~/components/common/Forms/NumericFormField/NumericFormField';
import styles from './styles';

class AreaRange extends Component {
  onChange = (fieldName, newValue) => {
    const { field, fieldApi, t } = this.props;
    const { values, touched } = fieldApi.getFormState();
    if (!touched || !touched[field]) return;

    if (touched[field].toArea && fieldName && fieldName[1] === 'fromArea') {
      const { toArea } = values[field];
      if (newValue > toArea) {
        const fromErrorMessage = t('searchResult:fromAreaShouldLessThanToArea');
        fieldApi.setError(fieldName, fromErrorMessage);
      } else {
        fieldApi.setError(fieldName, null);
        fieldApi.setError([field, 'toArea'], null);
      }
    } else if (touched[field].fromArea && fieldName && fieldName[1] === 'toArea') {
      const { fromArea } = values[field];
      if (newValue < fromArea) {
        const toErrorMessage = t('searchResult:toAreaShouldGreaterThanFromArea');
        fieldApi.setError(fieldName, toErrorMessage);
      } else {
        fieldApi.setError(fieldName, null);
        fieldApi.setError([field, 'fromArea'], null);
      }
    }
  }

  render() {
    const {
      field, fieldApi,
      t, fromAreaDefaultValue, toAreaDefaultValue,
    } = this.props;
    const { errors } = fieldApi.getFormState();
    let areaRangeError = {};
    if (errors && errors[field]) {
      areaRangeError = errors[field];
    }

    return (
      <section className={`row ${styles.areaRange}`}>
        <div className="col-md-6">
          <label className={`${styles.priceLabel}`}>{t('common:from')}</label>
          <div className="input-group">
            <Field
              field="fromArea"
              component={NumericFormField}
              defaultValue={fromAreaDefaultValue}
              onChange={this.onChange}
            />
            <span className="input-group-addon">m<sup>2</sup></span>
          </div>
          {
            areaRangeError.fromArea && (
              <div className={`${styles.error}`}>
                {areaRangeError.fromArea}
              </div>
            )
          }
        </div>

        <div className={classnames(`col-md-6 ${styles.paddingLeft}`)}>
          <label className={`${styles.priceLabel}`}>{t('common:to')}</label>
          <div className="input-group">
            <Field
              field="toArea"
              component={NumericFormField}
              defaultValue={toAreaDefaultValue}
              onChange={this.onChange}
            />
            <span className="input-group-addon">m<sup>2</sup></span>
          </div>
          {
            areaRangeError.toArea && (
              <div className={`${styles.error}`}>
                {areaRangeError.toArea}
              </div>
            )
          }
        </div>
      </section>
    );
  }
}

AreaRange.propTypes = {
  field: PropTypes.string.isRequired,
  fieldApi: PropTypes.object.isRequired,
  fromAreaDefaultValue: PropTypes.number,
  toAreaDefaultValue: PropTypes.number,
  t: PropTypes.func.isRequired,
};

AreaRange.defaultProps = {
  fromAreaDefaultValue: 0,
  toAreaDefaultValue: 0,
};

export default AreaRange;
