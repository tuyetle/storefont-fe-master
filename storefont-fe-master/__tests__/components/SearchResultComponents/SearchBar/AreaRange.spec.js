import React from 'react';
import renderer from 'react-test-renderer';
import AreaRange from '~/components/SearchResultComponents/SearchBar/AreaRange/AreaRange';

describe('(Component) AreaRange', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      field: 'sample',
      fieldApi: {
        getFormState() {
          return {};
        },
        setError: () => {},
      },
      fromAreaDefaultValue: 0,
      toAreaDefaultValue: 0,
      t: () => { },
    };

    component = <AreaRange {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

