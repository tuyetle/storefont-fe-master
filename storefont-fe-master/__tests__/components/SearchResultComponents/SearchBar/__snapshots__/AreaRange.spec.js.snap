// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AreaRange should render conrectly 1`] = `
<section
  className="row css-70fnwa"
>
  <div
    className="col-md-6"
  >
    <label
      className="css-legfcw"
    />
    <div
      className="input-group"
    >
      Field
      <span
        className="input-group-addon"
      >
        m
        <sup>
          2
        </sup>
      </span>
    </div>
  </div>
  <div
    className="col-md-6 css-17xsdqc"
  >
    <label
      className="css-legfcw"
    />
    <div
      className="input-group"
    >
      Field
      <span
        className="input-group-addon"
      >
        m
        <sup>
          2
        </sup>
      </span>
    </div>
  </div>
</section>
`;
